src/planner/expand_hypertable.c

```diff
@@ -396,8 +451,51 @@ transform_time_bucket_comparison(PlannerInfo *root, OpExpr *op)
 	if (strategy == BTGreaterStrategyNumber || strategy == BTGreaterEqualStrategyNumber)
 	{
 		/* column > value */
+		Expr *subst = value;
+
+		/* time_bucket (bucket_width, ts, timezone, origin, offset) */
+		if (tce->type_id == TIMESTAMPTZOID && list_length(time_bucket->args) == 5)
```

Comment:
If iam not wrong the 3rd argument in time_bucket means timezone. So why to check explicitly this condition ?
Will below work ?
if (list_length(time_bucket->args) == 5)
{
     Assert(tce->type_id == TIMESTAMPTZOID)
     ...

}

Reply:
The third argument is TIMEZONE just when the function have 5 arguments, have a look:
```sql
fabrizio=# SELECT proname, pg_get_function_arguments(oid) FROM pg_proc WHERE proname = 'time_bucket' AND array_length(proargnames, 1) >= 3 ORDER BY 2;
   proname   |                                                                              pg_get_function_arguments                                                                              
-------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 time_bucket | bucket_width bigint, ts bigint, "offset" bigint
 time_bucket | bucket_width integer, ts integer, "offset" integer
 time_bucket | bucket_width interval, ts date, "offset" interval
 time_bucket | bucket_width interval, ts date, origin date
 time_bucket | bucket_width interval, ts timestamp with time zone, "offset" interval
 time_bucket | bucket_width interval, ts timestamp with time zone, origin timestamp with time zone
 time_bucket | bucket_width interval, ts timestamp with time zone, timezone text, origin timestamp with time zone DEFAULT NULL::timestamp with time zone, "offset" interval DEFAULT NULL::interval
 time_bucket | bucket_width interval, ts timestamp without time zone, "offset" interval
 time_bucket | bucket_width interval, ts timestamp without time zone, origin timestamp without time zone
 time_bucket | bucket_width smallint, ts smallint, "offset" smallint
(10 rows)
```

Reply:
Ok, now i dont remember under what context i made this comment. Anyways if it works for you, then fine.